test/models/policy_host_test.rb

```diff
@@ -5,4 +5,65 @@
 class PolicyHostTest < ActiveSupport::TestCase
   should belong_to(:policy)
   should validate_presence_of(:host)
+
+  setup do
+    account1 = FactoryBot.create(:user).account
+    account2 = FactoryBot.create(:user).account
```

Comment:
This is hinting me that you're not testing what you should be testing. You have 2 separate (user) accounts for 2 separate policies and are trying to tangle assignments across users. That will never work as per pundit authorization...